torch/csrc/api/src/nn/modules/loss.cpp

```diff
@@ -5,17 +5,61 @@ namespace F = torch::nn::functional;
 namespace torch {
 namespace nn {
 
-L1LossImpl::L1LossImpl(const torch::nn::L1LossOptions& options_)
-    : options(options_) {}
+L1LossImpl::L1LossImpl(const L1LossOptions& options_) : options(options_) {}
 
 void L1LossImpl::reset() {}
 
 void L1LossImpl::pretty_print(std::ostream& stream) const {
-  stream << "torch::nn::L1Loss";
+  stream << "torch::nn::L1Loss()";
 }
 
 Tensor L1LossImpl::forward(const Tensor& input, const Tensor& target) {
-  return torch::l1_loss(input, target, options.reduction());
+  return F::l1_loss(input, target, options);
+}
+
+// ============================================================================
+
+KLDivLossImpl::KLDivLossImpl(const KLDivLossOptions& options_)
+    : options(options_) {}
+
+void KLDivLossImpl::reset() {}
+
+void KLDivLossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::KLDivLoss()";
+}
+
+Tensor KLDivLossImpl::forward(const Tensor& input, const Tensor& target) {
+  return F::kl_div(input, target, options);
+}
+
+// ============================================================================
+
+MSELossImpl::MSELossImpl(const MSELossOptions& options_) : options(options_) {}
+
+void MSELossImpl::reset() {}
+
+void MSELossImpl::pretty_print(std::ostream& stream) const {
+  stream << "torch::nn::MSELoss()";
+}
+
+Tensor MSELossImpl::forward(const Tensor& input, const Tensor& target) {
+  return F::mse_loss(input, target, options);
+}
+
+// ============================================================================
+
+BCELossImpl::BCELossImpl(const BCELossOptions& options_) : options(options_) {
+  register_parameter("weight", options.weight());
```

Comment:
Thanks for the catch! Yes I think we should move this into `reset()`

Reply:
Also it seems that it should be a buffer, not a parameter, based on the Python version:
```python
self.register_buffer('weight', weight)
```